asgiref/sync_iter.py

```diff
@@ -0,0 +1,64 @@
+import typing
```

Comment:
Please don't introduce typing just in one tiny place - I would like us to eventually add it properly all at once.